### DISCUSSION TITLE
How do I use @babel/generator to generate code that contains only single quotes?

### DISCUSSION THREAD

**Discussion by wood3n** [babel/babel#13989] [Q&A]
the `quotes` option was removed in `@babel/generator`, so how to generate a `StringLiteral` just use single quotes?

**Comment by fedeci** [comment]
You should apply some post-processor like `prettier` because `@babel/generator` output is not meant to be consistent (concerning code style).

**Reply by wood3n** [reply]
Thank you. That is the only way I think to solve this problem before, and now I have to use it.🤣

**Reply by fedeci** [reply]
It is definitely the most reliable too😅 But why would you like to beautify production code?

**Reply by wood3n** [reply]
> It is definitely the most reliable too😅 But why would you like to beautify production code?

Becauce my project needs to generate JS code whick is used in a VUE attribute, and it can only use single quotes😥, like below code, the `StringLiteral` in a `CallExpression` must use single quotes.
```vue
<div :id="generateKey('xxxx')"></div>
```

**Reply by fedeci** [reply]
I have really zero knowledge of vue. I am definitely not the right person to give you suggestions on it to avoid this "problem"😆

**Reply by JLHwung** [reply]
> my project needs to generate JS code whick is used in a VUE attribute

It seems to me you are consuming JS code as a source string that will be evaluated by the browser, in that case, prettier might be an overkill, you can call `JSON.stringify`:

```js
JSON.stringify(fs.readFile(generateCode))
```

**Comment by cha0s** [comment]
Here's a function you can use to create a single quoted string literal:

```js
const {stringLiteral} = require('@babel/types');
function stringLiteralSinglequote(value) {
  return {
    ...stringLiteral(value),
    extra: {rawValue: value, raw: `'${value}'`},
  };
}
```

Hope that helps!
